### ISSUE TITLE
measurement channel: Pointer or not

### ISSUE THREAD

**Issue by freeformz** [freeformz/shh#7]
Should the channel passed around to pollers and outputters be a chan *mm.Measurement or chan mm.Measurement ?

TL;DR It's currently *mm.Measurement. I'm not sure that's valid anymore.

This has the advantage that the reference (pointer) is passed on the channel, which I'm assuming cuts down on memory/gc, but if a poller/outputter decides to start keeping references to metrics around and for some reason those metrics get mutated, then that's potentially bad.

If I switch to a channel of mm.Measurement though, I can avoid that problem, but at the cost of copying each measurement passed on the channel.

I started with a pointer, since the normal use case it to just create them and pass them on the channel for something to output. But now things are having to convert counters to gauges, by keeping a reference to the last measurement of that name. From here things could start getting ugly.
